From 0cb5f56c07d112e5d5cc2e5145ddffdd30fb2c2c Mon Sep 17 00:00:00 2001
From: Andrzej Surdej <Andrzej_Surdej@comcast.com>
Date: Fri, 5 May 2023 17:17:21 +0200
Subject: [PATCH] comcast - DELIA-61287 - Disable WebAnimations interface

This fixes invalid CSS transition behaviour when target transform
is modified very close to animation end. As a result the next
animation was started from invalid point (start of previous animation
instead of current position).

See Github issue for reference:
https://github.com/WebPlatformForEmbedded/WPEWebKit/issues/1086
---
 Source/WebKit/Shared/WebPreferences.yaml | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Source/WebKit/Shared/WebPreferences.yaml b/Source/WebKit/Shared/WebPreferences.yaml
index 253e2aa56aff..c882a9558a06 100644
--- a/Source/WebKit/Shared/WebPreferences.yaml
+++ b/Source/WebKit/Shared/WebPreferences.yaml
@@ -1346,7 +1346,7 @@ CSSOMViewSmoothScrollingEnabled:
 
 WebAnimationsEnabled:
   type: bool
-  defaultValue: true
+  defaultValue: false
   humanReadableName: "Web Animations"
   humanReadableDescription: "Web Animations prototype"
   category: experimental
@@ -1354,7 +1354,7 @@ WebAnimationsEnabled:
 
 WebAnimationsCSSIntegrationEnabled:
   type: bool
-  defaultValue: true
+  defaultValue: false
   humanReadableName: "CSS Animations via Web Animations"
   humanReadableDescription: "Expose CSS Animations and CSS Transitions through getAnimations()"
   category: experimental
@@ -1362,7 +1362,7 @@ WebAnimationsCSSIntegrationEnabled:
 
 WebAnimationsCompositeOperationsEnabled:
   type: bool
-  defaultValue: DEFAULT_EXPERIMENTAL_FEATURES_ENABLED
+  defaultValue: false
   humanReadableName: "Web Animations composite operations"
   humanReadableDescription: "Support for the CompositeOperation enum and properties consuming it"
   category: experimental
@@ -1370,7 +1370,7 @@ WebAnimationsCompositeOperationsEnabled:
 
 WebAnimationsMutableTimelinesEnabled:
   type: bool
-  defaultValue: DEFAULT_EXPERIMENTAL_FEATURES_ENABLED
+  defaultValue: false
   humanReadableName: "Web Animations mutable timelines"
   humanReadableDescription: "Support for setting the timeline property of an Animation object"
   category: experimental
-- 
2.25.1

